File: 프로그래머스/0/120845. 주사위의 개수/주사위의 개수.py
def solution(box, n):
    answer = 1  # 주사위 개수를 저장할 변수

    for b in box:  # box 리스트의 각 요소(가로, 세로, 높이)에 대해 반복
        answer *= b // n  # 현재 차원에서 들어갈 수 있는 주사위 개수를 곱함

    return answer  # 최종적으로 들어갈 수 있는 주사위 개수 반환

# 테스트 예제 실행
print(solution([1, 1, 1], 1))    # 1
print(solution([10, 8, 6], 3))   # 12

File: 프로그래머스/0/120845. 주사위의 개수/주사위의 개수.java
public class Solution {

    public static int solution(int[] box, int n) {
        int answer = 1;

        // 각 차원(가로, 세로, 높이)에 대해 반복
        for (int i = 0; i < box.length; i++) {
            answer *= box[i] / n;  // 해당 방향에서 들어갈 수 있는 주사위 개수 곱셈
        }

        return answer;  // 전체 들어갈 수 있는 주사위 개수 반환
    }

    public static void main(String[] args) {
        System.out.println(solution(new int[]{1, 1, 1}, 1));    // 1
        System.out.println(solution(new int[]{10, 8, 6}, 3));   // 12
    }
}
